src/app/clusters/network-commissioning/network-commissioning.cpp

```diff
@@ -272,8 +294,19 @@ void Instance::HandleScanNetworks(HandlerContext & ctx, const Commands::ScanNetw
     MATTER_TRACE_EVENT_SCOPE("HandleScanNetwork", "NetworkCommissioning");
     if (mFeatureFlags.Has(NetworkCommissioningFeature::kWiFiNetworkInterface))
     {
+        if (!req.ssid.HasValue())
+        {
+            ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Protocols::InteractionModel::Status::InvalidCommand);
+            return;
+        }
+        const auto nullableSSID = req.ssid.Value();
+        ByteSpan ssid;
+        if (!nullableSSID.IsNull())
+        {
+            ssid = nullableSSID.Value();
+        }
         mAsyncCommandHandle = CommandHandler::Handle(&ctx.mCommandHandler);
-        mpDriver.Get<WiFiDriver *>()->ScanNetworks(req.ssid, this);
+        mpDriver.Get<WiFiDriver *>()->ScanNetworks(ssid, this);
```

Comment:
Unfortunately, the driver interface is not documented very well, so the implementations of it differ in how they handle the SSID.  For example, `src/platform/Ameba/NetworkCommissioningWiFiDriver.cpp` checks `ssid.data()` to decide whether we're doing a single-ssid scan or general scan, while `src/platform/EFR32/NetworkCommissioningWiFiDriver.cpp` checks `ssid.size()`.  If we checked the size here the difference would at least not be observable, but as it is they would reply differently to the same protocol message.

My recommendation would be to change the driver API to have separate functions for "scan for networks with the given SSID" and "scan for all networks" so the only place where we have to make that decision is here.